src/nix/all_deps.rs

```diff
@@ -0,0 +1,75 @@
+use std::fmt;
+
+use anyhow::{bail, Result};
+
+use super::devour_flake::DrvOut;
+
+/// Nix derivation path
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone, Hash)]
+pub struct Drv(pub String);
+
+/// Encompasses both derivation and derivation output paths
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone, Hash)]
+pub enum DrvPaths {
+    DrvOut(DrvOut),
+    Drv(Drv),
+}
```

Comment:
Also `DrvPaths`, what enum could better capture containing both `DrvOut` and `Drv`?

Reply:
Going by https://zero-to-nix.com/concepts/nix-store

```rust
pub enum StorePath {
  Drv(PathBuf),
  BuildOutput(PathBuff)
}
```

Reply:
I thought of that, but StorePath could include non-Drv or non-BuildOutput paths, like source paths for instance.

I guess its fine for now, I will rename it to StorePath.